ignite/engine/deterministic.py

```diff
@@ -176,7 +176,7 @@ class DeterministicEngine(Engine):
             in each iteration, and returns data to be stored in the engine's state.
     """
 
-    def __init__(self, process_function: Callable):
+    def __init__(self, process_function: Callable[["DeterministicEngine", Any], Any]):
```

Comment:
Looks like mypy is failing here, let's make the signature as `Callable[[Engine, Any], Any])`